File: boutique-hotel-technikum/src/main/java/Model/Booking.java
package Model;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Getter @Setter @NoArgsConstructor
public class Booking {
    @Id
    @Column(name = "id")
    private int id;
    @Column(name = "title")
    private String title;
    @Column(name = "additional_notes")
    private String additional_notes;
    @Column(name = "customer_id")
    private int customer_id;
    @Column(name = "room_id")
    private String room_id;
    @Column(name = "created_at")
    private String created_at;
    @Column(name = "booking_start")
    private String booking_start;
    @Column(name = "booking_end")
    private String booking_end;
    @Column(name = "breakfast")
    private boolean breakfast;
    @Column(name = "total_cost")
    private double total_cost;
    @Column(name = "cancelled")
    private boolean cancelled;

    public Booking(String title, String additional_notes, int customer_id, String room_id, String created_at, String booking_start, String booking_end, boolean breakfast, double total_cost, boolean cancelled){
        this.title = title;
        this.additional_notes = additional_notes;
        this.customer_id = customer_id;
        this.room_id = room_id;
        this.created_at = created_at;
        this.booking_start = booking_start;
        this.booking_end = booking_end;
        this.breakfast = breakfast;
        this.total_cost = total_cost;
        this.cancelled = cancelled;
    }
}

File: boutique-hotel-technikum-app/src/stores/booking.ts
import { defineStore, acceptHMRUpdate } from 'pinia'
import { useUserStore } from './user'
import axios from "axios";
import {book} from "ionicons/icons";

// Define the state interface
interface BookingState {
    id: number;
    title: string;
    additionalNotes: string;
    customerId: number;
    roomId: number;
    createdAt: string;
    bookingStart: string,
    bookingEnd: string,
    breakfast: boolean,
    totalCost: number,
    cancelled: boolean
}

const apiUrl: string =  "http://localhost:8080/api/hotel/v1/booking";
export const useBookingStore = defineStore({
    id: 'booking',

    state: (): BookingState => ({
        id: 0,
        title: "string",
        additionalNotes: "string",
        customerId: 0,
        roomId: 0,
        createdAt: "string",
        bookingStart: "string",
        bookingEnd: "string",
        breakfast: false,
        totalCost: 0,
        cancelled: false
    }),
    getters: {
    },
    actions: {
        async createBooking(title: string,additionalNotes: string, customerId: number, roomId: number, bookingStart: string,bookingEnd: string,breakfast: boolean, totalCost: number) {
            try {
                const response = await axios.post<BookingState>(`${apiUrl}`,{
                    title: title,
                    additionalNotes: additionalNotes,
                    customerId: customerId,
                    roomId: roomId,
                    bookingStart: bookingStart,
                    bookingEnd: bookingEnd,
                    breakfast: breakfast,
                    totalCost: totalCost,

                });
                const booking = response.data;
                console.log(booking)
                this.id = booking.id;
                this.title = booking.title;
                this.additionalNotes = booking.additionalNotes;
                this.customerId = booking.customerId;
                this.roomId = booking.roomId;
                this.createdAt = booking.createdAt;
                this.bookingStart = booking.bookingStart;
                this.bookingEnd = booking.bookingEnd;
                this.breakfast = booking.breakfast;
                this.totalCost = booking.totalCost;
                this.cancelled = booking.cancelled;
                return booking
            } catch (error) {
                console.error('Error creating booking data:', error);
            }
        },
    },
})

